mailpoet/mailpoet
# Migrations [MAILPOET-4466]

## Description

Implements new `MailPoet\Migrations` and fully replaces`MailPoet\Config\Migrator` (and a part of `MailPoet\Config\Populator`).

The new migrator is executed in the same places as the previous one (in activator). It also adds a few commands for development:

```bash
./do migrations:new     # creates a new migration file
./do migrations:status  # displays detailed info about migrations (must be run in Docker)
./do migrations:run     # runs all new migrations (must be run in Docker)
```

## Code review notes

I recommend looking commit-by-commit, hopefully it's clear that way.

## QA notes

Please test plugin upgrade from older versions such as `3.26.0`, `3.31.1`, `3.38.1`, `3.42.0`, `3.46.13`, `3.47.6`, `3.48.1`, `3.49.1`, `3.60.4`, `3.63.0`, `3.67.0`, `3.73.0`, `3.73.2`, `3.74.2`, `3.75.1`, `3.77.1`, `3.78.0`, `3.99.0`, `3.91.1`.

It is not necessary to test all of them! Two or three random picks should do the job. Maybe not so random, in fact — I recommend including one of the older versions in the mix, that will execute most of the database migration logic.

Note that automations migration is not added yet as we still have the feature flat in place.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4466]

## After-merge notes

Ping @JanJakes to inform the team about new migrations.


[MAILPOET-4466]: https://mailpoet.atlassian.net/browse/MAILPOET-4466?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
